crates/ruff/src/rules/tryceratops/rules/verbose_raise.rs

```diff
@@ -88,17 +91,32 @@ pub(crate) fn verbose_raise(checker: &mut Checker, handlers: &[ExceptHandler]) {
                 visitor.visit_body(body);
                 visitor.raises
             };
-            for (exc, cause) in raises {
-                if cause.is_some() {
+            for raise in raises {
+                if raise.cause.is_some() {
                     continue;
                 }
-                if let Some(exc) = exc {
+                if let Some(exc) = raise.exc.as_ref() {
                     // ...and the raised object is bound to the same name...
-                    if let Expr::Name(ast::ExprName { id, .. }) = exc {
+                    if let Expr::Name(ast::ExprName { id, .. }) = exc.as_ref() {
                         if id == exception_name.as_str() {
-                            checker
-                                .diagnostics
-                                .push(Diagnostic::new(VerboseRaise, exc.range()));
+                            let mut diagnostic = Diagnostic::new(VerboseRaise, exc.range());
+                            if checker.patch(diagnostic.kind.rule()) {
+                                diagnostic.set_fix(Fix::automatic_edits(
```

Comment:
Can this be simplified to: `Edit::range_replacement("raise".to_string(), raise.range())`?

Reply:
I think it should be "suggested" because this rule can have some false positives. I assume it gets this wrong:

```python
except Foo as e:
  e = 1
  raise e
```

Reply:
Oh yes, I had that case in mind and it does gets it wrong. I was wondering if our semantic model can resolve such references (I can't find anything on that). I'll make this a suggested fix.

Reply:
Could we change to `Edit::range_replacement("raise".to_string(), raise.range())` too, or does that not work?

Reply:
Oh we need to delete `as e` too, I see. Does this rule verify that `e` isn't used in the exception body? Like:

```python
except Foo as e:
  print(e)
  raise e
```

In that case, we can change `raise e` to `raise`, but we can't remove `as e`.

Reply:
No, we don't really check that case. I don't think we can use the semantic model to verify such instances so I'll probably have to check either using a `StatementVisitor` or `any_over_stmt`.

Reply:
I think this fix should _only_ remove `e` in `raise e`. If the variable is unused, it'll get marked in the next pass as an unused variable and removed via the unused variable autofix.

Reply:
Oh right, that makes sense. I'll just add a test case for it then and remove the first edit.

Reply:
Sounds good. Eventually we can improve this rule to handle it properly in the first place.

Reply:
Hmm, this is interesting: Ruff is detecting that the variable `e` is unused in the last test case which seems incorrect:

```python
def bad_that_needs_recursion_2():
    try:
        process()
    except MyException as e:  # F841: Local variable `e` is assigned to but never used
        logger.exception("process failed")
        if True:

            def foo():
                raise e  # F821: Undefined name `e`
```